include/seqan3/alignment/pairwise/policy/find_optimum_policy.hpp

```diff
@@ -0,0 +1,123 @@
+    using find_in_last_column_t = std::false_type;
+};
+
+/*!\brief A policy to determine the optimum of the dynamic programming matrix.
+ * \ingroup alignment_policy
+ * \tparam derived_t        The type of the derived class.
+ * \tparam core_policy_t    The type of the core alignment policy which also might need access to the member functions.
+ * \tparam traits_type      A traits type that determines which cells should be considered for the optimum.
+ *                          Defaults to seqan3::detail::default_find_optimum_trait.
+ */
+template <typename derived_t, typename core_policy_t, typename traits_type = default_find_optimum_trait>
+class find_optimum_policy
+{
+
+private:
+
+    //!\brief Befriends the derived class to grant it access to the private members.
+    friend derived_t;
+    //!\brief Befriends the core_policy_t which is still a seqan3::detail::deferred_crtp_base and must be resolved
+    //!\      accordingly.
+    friend invoke_deferred_crtp_base<core_policy_t, derived_t>;
+
+    /*!\name Constructor, destructor and assignment
+     * \brief Defaulted all standard constructor.
+     * \{
+     */
+    constexpr find_optimum_policy()                                        = default;
+    constexpr find_optimum_policy(find_optimum_policy const &)             = default;
+    constexpr find_optimum_policy(find_optimum_policy &&)                  = default;
+    constexpr find_optimum_policy & operator=(find_optimum_policy const &) = default;
+    constexpr find_optimum_policy & operator=(find_optimum_policy &&)      = default;
+    ~find_optimum_policy()                                                 = default;
+    //!}
+
+    /*!\brief Checks every cell of the dynamic programming matrix.
+     * \tparam score_t  The type of the score.
+     * \param[in]     val       The current value.
+     * \param[in,out] optimum   The current optimum to compare with.
+     */
+    template <typename score_t>
+    constexpr void check_score([[maybe_unused]] score_t const val,
+                               [[maybe_unused]] score_t & optimum) const noexcept
+    {
+        if constexpr (traits_type::find_in_every_cell_t::value)
+            optimum = std::max(optimum, val);
+    }
+
+    /*!\brief Checks the last cell of the last row of the dynamic programming matrix.
+     * \tparam score_t  The type of the score.
+     * \param[in]     val       The current value.
+     * \param[in,out] optimum   The current optimum to compare with.
+     */
+    template <typename score_t>
+    constexpr void check_score_last_row([[maybe_unused]] score_t const val,
+                                        [[maybe_unused]] score_t & optimum) const noexcept
+    {
+        if constexpr (traits_type::find_in_last_row_t::value)
+            optimum = std::max(optimum, val);
+    }
+
+    /*!\brief Checks the complete last column for the optimal score.
+     * \tparam rng_t    The type of the last column. Must be at least an input range.
```

Comment:
Mhm if it needs to be an input range, you could also say `models std::InputRange` and require it?

Reply:
It is actually required to be a bidirectional range and I added a concept.